fairseq/sequence_generator.py

```diff
@@ -335,10 +346,13 @@ def _generate(
             cand_bbsz_idx = cand_beams.add(bbsz_offsets)
 
             # finalize hypotheses that end in eos
+            # Shape of eos_mask: (batch size, beam size)
```

Comment:
nice, thanks for all these comments. This code is already so complicated, every bit of documentation helps :)